### PR TITLE
Add username validation for alphanumeric characters and underscores only

### PR DESCRIPTION
**Pull request by ManukMinasyan**
Restricts usernames to only alphanumeric characters and underscores.
Additionally, tests have been added to verify the correctness of these validation rules.

### CODE REVIEW

File: tests/Unit/Rules/Username.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    $rule = new Username();
+
+    $invalidUsernames = [
+        'invalid username',
+        'invalid@username',
+        'username!',
+        '12345$',
+        '-',
+        '   ',
+    ];
```
Comment (MrPunyapal) [inline tests/Unit/Rules/Username.php:34]: Can we use ->with instead of this?

File: tests/Unit/Rules/Username.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+use App\Rules\Username;
+
+test('username validation passes for valid usernames', function () {
+    $rule = new Username();
+
+    $validUsernames = [
+        'valid_username',
+        'User123',
+        '_underscore',
+    ];
```
Comment (MrPunyapal) [inline tests/Unit/Rules/Username.php:15]: Can we use ->with instead of this?

**Review by MrPunyapal — CHANGES_REQUESTED** [review]

File: tests/Unit/Rules/Username.php (hunk 1 above)
Comment (ManukMinasyan) [reply tests/Unit/Rules/Username.php]: Done

**Comment by MrPunyapal** [comment]
You forgot to add Test postfix.

**Comment by ManukMinasyan** [comment]
> You forgot to add Test postfix.

I didn't understand what was being discussed.

**Comment by MrPunyapal** [comment]
`Username.php` to `UsernameTest.php`

**Comment by ManukMinasyan** [comment]
> `Username.php` to `UsernameTest.php`

Done

**Review by MrPunyapal — APPROVED** [review]

**Pull request merged**